lib/active_model_serializers/adapter/json_api/relationship.rb

```diff
@@ -34,6 +34,11 @@ def as_json
 
         private
 
+        def ensure_valid!(json)
```

Comment:
I need to submit a PR for just nil input. My issue for that was that the _wrong_ error was being thrown.